packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts

```diff
@@ -38,27 +40,114 @@ const GlobalClassNames = {
   custom: 'ms-Panel--custom'
 };
 
-const panelSize = {
-  width: {
-    xs: '272px',
-    sm: '340px',
-    md: '643px',
-    lg: '940px'
+const panelWidth = {
+  full: '100%',
+  auto: 'auto',
+  xs: 272,
+  sm: 340,
+  md1: 592,
+  md2: 644,
+  lg: 940
+};
+
+const panelMargin = {
+  auto: 'auto',
+  none: 0,
+  md: 48,
+  lg: 428,
+  xl: 176
+};
+
+const smallPanelSelectors = {
+  ['@media (min-width: ' + ScreenWidthMinMedium + 'px)']: {
+    width: panelWidth.sm
+  }
+};
+
+const mediumPanelSelectors = {
+  ['@media (min-width: ' + ScreenWidthMinLarge + 'px)']: {
+    width: panelWidth.md1
   },
-  margin: {
-    md: '48px',
-    lg: '428px',
-    xl: '176px'
+  ['@media (min-width: ' + ScreenWidthMinXLarge + 'px)']: {
+    width: panelWidth.md2
   }
 };
 
+const largePanelSelectors = {
+  ['@media (min-width: ' + ScreenWidthMinUhfMobile + 'px)']: {
+    left: panelMargin.md,
+    width: panelWidth.auto
+  },
+  ['@media (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
+    left: panelMargin.lg
+  }
+};
+
+const largeFixedPanelSelectors = {
+  ['@media (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
+    left: panelMargin.auto,
+    width: panelWidth.lg
+  }
+};
+
+const extraLargePanelSelectors = {
+  ['@media (min-width: ' + ScreenWidthMinXXLarge + 'px)']: {
+    left: panelMargin.xl
+  }
+};
+
+const getPanelBreakpoints = (type: PanelType): { [x: string]: IStyle } | undefined => {
```

Comment:
Extracted all the breakpoints logic in a separate function to easily follow the fall down of the styles in different breakpoint for different `Panel` types.